File: frontend/src/services/electionService.js
import api from "./api"



export const getElections = async () => {
  const response = await api.get(
    "/elections/elections/"
  )

  return response.data
}



export const createElection = async (
  formData
) => {
  const response = await api.post(
    "/elections/elections/",
    formData
  )

  return response.data
}



export const updateElection = async (
  electionId,
  formData
) => {
  const response = await api.put(
    `/elections/elections/${electionId}/`,
    formData
  )

  return response.data
}



export const deleteElection = async (
  electionId
) => {
  const response = await api.delete(
    `/elections/elections/${electionId}/`
  )

  return response.data
}

File: VotingSystem/mobile/lib/services/election_service.dart
import '../api/api_client.dart';
import '../models/election_model.dart';

class ElectionService {
  static Future<List<ElectionModel>>
      getElections() async {

    final response =
        await ApiClient.dio.get(
      '/election/elections/',
    );

    print(response.data);

    final data = response.data;

    List elections = [];

    if (data is List) {
      elections = data;
    } else if (data is Map &&
        data.containsKey('results')) {
      elections = data['results'];
    }

    return elections
        .map(
          (e) => ElectionModel.fromJson(e),
        )
        .toList();
  }
}